### ISSUE TITLE
do not vendor net-http-presistent

### ISSUE THREAD

**Issue by grosser** [hashicorp/vault-ruby#111]
no I have all these errors whenever I start my app ... not fun :(

```
vendor/bundle/ruby/2.3.0/gems/vault-0.7.0/lib/vault/vendor/net/http/persistent.rb:217: warning: already initialized constant Net::HTTP::Persistent::RETRIED_EXCEPTIONS
vendor/bundle/ruby/2.3.0/gems/net-http-persistent-2.9.4/lib/net/http/persistent.rb:212: warning: previous definition of RETRIED_EXCEPTIONS was here
```

@evanphx

**Comment by grosser** [comment]
... or at least require the original ... but even that would be pretty hacky and could lead to weird bugs

**Comment by sethvargo** [comment]
Hi @grosser 

Those are just warnings that can safely be ignored. The reason we vendor is because it makes it challenging to create a constraint that doesn't block upstream in vault-rails. net-http-persistent is a very popular library, and it makes it challenging to find a constraint that will satisfy everyone's needs.

**Comment by grosser** [comment]
This cannot be 'safely ignored' ... because it is not 'safe' :D 

instead of saying "this works with any version of http-persistent" you chose to "show warnings and then use any version of http-persistent" (because it is overwritten)

afaik safe options are:
 - declare '>= 0' dependency and test accordingly
 - declare stricter dependency and possibly break old versions of vault-rails
 - vendor by putting the whole library into the `Vault` namespace

**Comment by grosser** [comment]
... please reopen

**Comment by evanphx** [comment]
This has been fixed in master, I'll be releasing 0.7.1 shortly with the fixes.
